services/vidispine/vsapi/chapter.go

```diff
@@ -30,6 +25,7 @@ func (c *Client) GetChapterMeta(itemVXID string, inTc, outTc float64) (map[strin
 			continue
 		}
 
+		// TODO: @KillerX can you document why we need to do this? e.g. by extracting into a function
 		for _, field := range clip.Terse {
```

Comment:
Essentially we filter out chapters that are not part of the subclip:

```
+-------------------------------------------------------------------------------------------+
|                         Full Asset                                                        |
+-------------------------------------------------------------------------------------------+
+-----------------------------+                                                              
|      Chapter 1              |                                                              
+-----------------------------+                                                              
                   +--------------------------------+                                        
                   |       Chapter 2                |                                        
                   +--------------------------------+                                        
                                       +-------------------------------------+               
                                       |           Chapter 3                 |               
                                       +-------------------------------------+               
                             +--------------------------------------------------------------+
                             |                Chapter 4                                     |
                             +--------------------------------------------------------------+
                                                                                             
                                  +---------------------------------+                        
                                  |        CLIP                     |                        
                                  +---------------------------------+                        
```

In this case you only want Chapter 2-4 and not chapter 1.

In addition the chapter start and end times get cut to not extend beyond the clip start and end time.

Reply:
moved it to a func "trimTimecodesToBeWithinRange"